doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -279,6 +279,7 @@ Field         | Type       | Meaning
 `worker_id`   | [`bigint`] | The ID of the worker thread servicing the peek.
 `duration_ns` | [`bigint`] | The upper bound of the bucket in nanoseconds.
 `count`       | [`bigint`] | The (noncumulative) count of peeks in this bucket.
+`sum`         | [`bigint`] | The (noncumulative) sum of peek durations in this bucket, or `NULL` in the unlikely event it has overflowed.
```

Comment:
With "unlikely" you mean "practically impossible" here, right? If I'm not mistaken, overflowing this value would require a sustained ~585 billion peeks per second for a whole year. We never let replicas run for a whole year and I'm sure we cannot handle nearly as many peeks per second.

Reply:
I think your math is off -- probably either you calculated with seconds (instead of nanoseconds) or assumed this was a count, not a sum. In fact it is a sum of nanoseconds.

If each peek lasts a day, this will overflow after about 200,000 peeks. That's _still_ probably impossible, but it's less far-fetched than what you suggest. I don't want to have to think forever about whether we are introducing a way to cause long-running peeks and therefore make this overflow.

However, even given these considerations, I was a bit conflicted about whether to make the column nullable since indeed, an overflow can't happen in the current system. What do you think?

Reply:
You are right, I for some reason assumed this was a peek count (which is the row above it). Sorry for the noise!

> However, even given these considerations, I was a bit conflicted about whether to make the column nullable since indeed, an overflow can't happen in the current system. What do you think?

Since this is an mz_internal catalog item, we are fine making this column non-nullable now, since we are allowed to make it nullable again later. The only danger is that we might stabilize it one day, and then forget about thinking about nullability again.

I don't really mind keeping it nullable, actually. I was thinking we maybe don't need to document this edge-case, if it's not going to happen anyway. But since this is not as unlikely as I initially thought, I'd be fine with keeping it.

Reply:
I will add a comment to the code summarizing this discussion, to force us to think about it again if we stabilize.